language/transaction-builder/src/lib.rs

```diff
@@ -19,6 +19,15 @@ use stdlib::transaction_scripts::{
 #[cfg(any(test, feature = "fuzzing"))]
 use vm::file_format::{Bytecode, CompiledScript};
 
+fn validate_auth_key_prefix(auth_key_prefix: &[u8]) {
+    let auth_key_prefix_length = auth_key_prefix.len();
+    assert!(
+        auth_key_prefix_length == 0 || auth_key_prefix_length == 16,
```

Comment:
Let's replace 16 with `AUTHENTICATION_KEY_LENGTH - ADDRESS_LENGTH` (or create a constant for prefix expected size)